examples/nlp/machine_translation/create_tarred_monolingual_dataset.py

```diff
@@ -19,8 +19,10 @@
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description='NMT dataset pre-processing')
-    parser.add_argument('--tokenizer_model', type=str, required=True, help='Path to tokenizer model')
-    parser.add_argument('--tokenizer_name', type=str, default='yttm', help='BPE Tokenizer Name, Options: [yttm]')
+    parser.add_argument(
+        '--tokenizer_name', type=str, default='yttm', help='Supports yttm, sentencepiece and HuggingFace tokenizers',
+    )
+    parser.add_argument('--tokenizer_model', type=str, default=None, help='Path to tokenizer model')
```

Comment:
I think having the tokenizer_model is necessary and it isn't possible to train a tokenizer on the provided data right now correct? If so, I think we should make this a required argparse field.

Reply:
The reason for removing `required` flag is to support HF tokenisers out of the box. They are usually created by passing corresponding HF model name to `tokenizer_name`.

Alternative design choice is to pass something like `hf` or `hugging_face` as `tokenizer_name` and expecting `tokenizer_model` to be the name of the corresponding HF model.